feat: add heading styles for o3-editroial-typography (#1603)
Financial-Times/origami: components/o3-editorial-typography/README.md

# o3-editorial-typography

Typographic styles for editorial content.

- [o3-editorial-typography](#o3-editorial-typography)
  - [Overview](#overview)
  - [Markup](#markup)
    - [Heading styles](#heading-styles)
      - [HTML](#html)
      - [JSX](#jsx)
  - [Migration](#migration)
  - [Contact](#contact)
  - [Licence](#licence)

## Overview

> O3 editorial typography is keeping typography consistent and sticking to logical hierarchies ensures that page elements are clear and easily recognisable when scanning the page. Lead with Mobile-First (also applies to FT Apps. For example; FT Apps, FT Edit). `o3-editorial-typography` only supports `core` and `sustainable-view` brands.

Reviewer: This strays into design guidelines, which would be better placed on the site. How about?

```suggestion
`o3-editorial-typography` provides typographic styles for editorial content. It supports the following brands:
- `core`
-`sustainable-view`

For UI typography see `o3-typography`.
```

## Markup

o3-editorial-typography supports [JSX templates for React users](#jsx), or direct HTML. We recommend using JSX where possible.

### Heading styles

Heading styles are available in 5 different types.

| Type           | Selector                                 |
| -------------- | ---------------------------------------- |

> | Headline Large | .o3-editorial-typography--headline-large |
> | Headline       | .o3-editorial-typography--headline       |
> | Chapter        | .o3-editorial-typography--chapter        |
> | Subheading     | .o3-editorial-typography--subheading     |
> | Label          | .o3-editorial-typography--label          |

Reviewer: As these aren't modifier classes anymore we should replace the `--` for just `-`.
```suggestion
| Headline Large | .o3-editorial-typography--headline-large |
| Headline       | .o3-editorial-typography--headline       |
| Chapter        | .o3-editorial-typography--chapter        |
| Subheading     | .o3-editorial-typography--subheading     |
| Label          | .o3-editorial-typography--label          |
```

#### HTML

```html
<h1 class="o3-editorial-typography--headline-large">
 Large headline
</h1>

<h1 class="o3-editorial-typography--headline">
 Headline
</h1>

<h2 class="o3-editorial-typography--chapter">
 Chapter
</h2>

<h3 class="o3-editorial-typography--subheading">
 Subheading
</h3>

<h3 class="o3-editorial-typography--label">Label</h3>
```

#### JSX

> Same heading styles can be used in JSX environments as well by importing `<Headline>` component and applying `type` modifier. The component is exported as commonjs module as well as es module. Depending your project setup, you can import the component from `/cjs` or `/esm` directory.

Reviewer: ```suggestion
The same heading styles can be used in JSX environments as well by importing `<Headline>` component and applying the `type` modifier. The component is exported as a commonjs module as well as an es module. Depending your project setup, you can import the component from `/cjs` or `/esm` directory.
```

☝️ For the site demos I've successfully imported JSX through `@financial-times/o3-editorial-typography`, allowing Vite to select the appropriate export. I'm thinking should we..?

1. Use that style of import in the REDME, because it's the simplest?
2. Document somewhere generic in technical o3 guides to use a specific cjs or esm given issues resolving the right one?

Author: I think for this one we decided to document this somewhere else as a more generic guideline and we have not done it yet so I am including it with the component once we have more documentation this part can go away.

```tsx
import { Headline } from '@financial-times/o3-editorial-typography/cjs';

<Headline type="headline-large">Large headline</Headline>
<Headline type="headline">Headline</Headline>
<Headline type="chapter">Chapter</Headline>
<Headline type="subheading">Subheading</Headline>
<Headline type="label">Label</Headline>

```

`<Headline>` props:
| Prop | Type | Default | Description |
| --- | --- | --- | --- |
| type | `headline-large` \| `headline` \| `chapter` \| `subheading` \| `label` | `headline` | Type of the headline.|
| theme | `standard` \| `inverse` | '' | Theme of the headline. |

## Theme modifiers

> `o3-editorial-typography` also supports `inverse` theme. Adding `data-o3-theme="inverse"` to the wrapper element or at the element itself will change the color of the text to white.

Reviewer: [British english for docs, American for code ](https://origami.ft.com/documentation/principles/tone-and-language/#use-british-english-for-documentation)

```suggestion
`o3-editorial-typography` also supports the `inverse` theme. Adding `data-o3-theme="inverse"` to the wrapper element or at the element itself will change the colour of the text to white.
```

```html
<h3

>  class="o3-editorial-typography o3-editorial-typography--label"

Reviewer: ```suggestion
 class="o3-editorial-typography--label"
```
